core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricUtils.scala

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.core.env.StreamUtilities
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import spray.json._
+
+case class TokenServiceConfig(tokenServiceEndpoint: String,
+                              clusterType: String,
+                              clusterName: String,
+                              sessionToken: String)
+
+object TokenServiceConfigProtocol extends DefaultJsonProtocol {
+  implicit val TokenServiceConfigFormat: RootJsonFormat[TokenServiceConfig] = jsonFormat4(TokenServiceConfig)
+}
+
+import TokenServiceConfigProtocol._
+
+object FabricUtils {
+  //private var TridentContext: Map[String, String] = Map[String, String]()
+
+  def getFabricContext: Map[String, String] = {
+    try {
+      val linesContextFile = StreamUtilities.usingSource(scala.io.Source.fromFile(FabricConstants.ContextFilePath)) {
+        source => source.getLines().toList
+      }.get
+
+      val tokenServiceConfig = StreamUtilities.usingSource(scala.io.Source.fromFile
+      (FabricConstants.TokenServiceFilePath)) {
+        source => source.mkString.parseJson.convertTo[TokenServiceConfig]
+      }.get
+
+      val tridentContext: Map[String, String] = linesContextFile
+        .filter(line => line.split('=').length == 2)
+        .map { line =>
+          val Array(k, v) = line.split('=')
+          (k.trim, v.trim)
+        }.toMap
+        .++(Seq(
+          (FabricConstants.SynapseTokenServiceEndpoint, tokenServiceConfig.tokenServiceEndpoint),
+          (FabricConstants.SynapseClusterType, tokenServiceConfig.clusterType),
+          (FabricConstants.SynapseClusterIdentifier, tokenServiceConfig.clusterName),
+          (FabricConstants.TridentSessionToken, tokenServiceConfig.sessionToken)
+        ).toMap)
+
+      tridentContext
+    } catch {
+      case e: NullPointerException =>
```

Comment:
Don’t catch errors just throw them if something is wrong. Catch at the last possible moment and only if necessary